src/crypto/internal/boring/boring.go

```diff
@@ -126,23 +149,34 @@ func PanicIfStrictFIPS(msg string) {
 }
 
 func NewOpenSSLError(msg string) error {
-	var b strings.Builder
 	var e C.ulong
-
-	b.WriteString(msg)
-	b.WriteString("\nopenssl error(s):\n")
-
-	for {
-		e = C._goboringcrypto_internal_ERR_get_error()
-		if e == 0 {
-			break
+	message := fmt.Sprintf("\n%v\nopenssl error(s):", msg)
+	if openSSLVersion() >= OPENSSL_VERSION_3_0_0 {
+		for {
+			var buf [256]C.char
+			var file, fnc, data *C.char
+			var line, flags C.int
+			e = C._goboringcrypto_internal_ERR_get_error_all(&file, &line, &fnc, &data, &flags)
+			if e == 0 {
+				break
+			}
+
+			C._goboringcrypto_internal_ERR_error_string_n(e,(*C.uchar)(unsafe.Pointer (&buf[0])), 256)
+			message = fmt.Sprintf(
+				"%v\nfile: %v\nline: %v\nfunction: %v\nflags: %v\nerror string: %s\n",
+				message,C.GoString(file), line, C.GoString(fnc), flags, C.GoString(&(buf[0])))
+		}
+	} else {
+		for {
+			var buf [256]C.char
+			C._goboringcrypto_internal_ERR_error_string_n(e,(*C.uchar)(unsafe.Pointer (&buf[0])), 256)
```

Comment:
`e` is unassigned in this loop, as `e = C._goboringcrypto_internal_ERR_get_error()` is gone? Also `message` set at line 153 is overwritten?

Reply:
Great catch, deleted by mistake while refactoring.  Thanks!